synapseclient/models/table_components.py

```diff
@@ -482,6 +514,391 @@ def __repr__(self) -> str:
+
+    @classmethod
+    def fill_from_dict(cls, data: Dict[str, Any]) -> "SelectColumn":
+        """Create a SelectColumn from a dictionary response."""
+        return cls(
+            name=data.get("name", ""),
+            column_type=ColumnType(data.get("columnType", None)),
+            id=data.get("id"),
+        )
+
+
+@dataclass
+class QueryNextPageToken:
+    """
+    Token for retrieving the next page of query results.
+
+    This result is modeled from: <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/QueryNextPageToken.html>
+    """
+
+    concrete_type: Optional[str] = None
+    """The concrete type of this object"""
+
+    entity_id: Optional[str] = None
+    """The ID of the entity (table/view) being queried"""
+
+    token: Optional[str] = None
+    """The token for the next page."""
+
+    @classmethod
+    def fill_from_dict(cls, data: Dict[str, Any]) -> "QueryNextPageToken":
+        """Create a QueryNextPageToken from a dictionary response."""
+        return cls(
+            concrete_type=data.get("concreteType"),
+            entity_id=data.get("entityId"),
+            token=data.get("token"),
+        )
+
+
+@dataclass
+class RowSet:
+    """
+    Represents a set of row of a TableEntity.
+
+    This result is modeled from: <https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/table/RowSet.html>
+    """
+
+    concrete_type: Optional[str] = None
+    """The concrete type of this object"""
+
+    table_id: Optional[str] = None
+    """The ID of the TableEntity than owns these rows"""
+
+    etag: Optional[str] = None
+    """Any RowSet returned from Synapse will contain the current etag of the change set.
+    To update any rows from a RowSet the etag must be provided with the POST."""
+
+    headers: Optional[List[SelectColumn]] = None
+    """The list of SelectColumns that describes the rows of this set."""
+
+    rows: Optional[List[Row]] = None
+    """The Rows of this set. The index of each row value aligns with the index of each header."""
+
+    @classmethod
+    def cast_row(
+        cls, row: Dict[str, Any], headers: List[Dict[str, Any]]
+    ) -> Dict[str, Any]:
+        row["values"] = Row.cast_values(row["values"], headers)
+        return row
+
+    @classmethod
+    def cast_row_set(cls, rows: List[Row], headers: List[Dict[str, Any]]) -> List[Row]:
+        rows = [cls.cast_row(row, headers) for row in rows]
+        return rows
+
+    @classmethod
+    def fill_from_dict(cls, data: Dict[str, Any]) -> "RowSet":
+        """Create a RowSet from a dictionary response."""
+        headers = data.get("headers", None)
+        rows = data.get("rows", None)
+
+        # Handle headers
+        headers_data = data.get("headers")
+        if headers_data and isinstance(headers_data, list):
+            headers = [SelectColumn.fill_from_dict(header) for header in headers_data]
+
+        # Handle rows
```

Comment:
This section was added to ensure backward compatibility with existing functions See how casting works in the original function here: https://github.com/Sage-Bionetworks/synapsePythonClient/blob/develop/synapseclient/table.py#L1669